packages/aragon-wrapper/src/index.js

```diff
@@ -197,31 +199,114 @@ export default class Aragon {
    * @return {void}
    */
   initNotifications () {
-    this.notifications = new BehaviorSubject(
-      this.cache.get('notifications', [])
-    )
-      .scan((notifications, notification) => notifications.concat(notification))
-      .do((notifications) => this.cache.set(notifications))
+    // If the cached notifications doesn't exist or isn't an array, set it to an empty one
+    let cached = this.cache.get('notifications')
+    if (!Array.isArray(cached)) {
+      cached = []
+    } else {
+      // Set up acknowledge for unread notifications
+      cached.forEach(notification => {
+        if (notification && !notification.read) {
+          notification.acknowledge = () => this.acknowledgeNotification(notification.id)
+        }
+      })
+    }
+
+    this.notifications = new BehaviorSubject(cached)
+      .scan((notifications, { modifier, notification }) => modifier(notifications, notification))
+      .do((notifications) => this.cache.set('notifications', notifications))
+      .publishReplay(1)
```

Comment:
Adding `.publishReplay()` seems to turn the `BehaviourSubject` into a multicasted observable, such that all observers (even those that subscribe later) receive the same values (rather than only the last one sent through the subject). Not 100% sure this is correct, though.

Reply:
`publishReplay` wraps the observable in a `ReplaySubject`, but a `BehaviorSubject` is already a `ReplaySubject`, so it's pretty verbose to have it.

http://reactivex.io/rxjs/test-file/spec-js/subjects/BehaviorSubject-spec.js.html#lineNumber106

Reply:
The behaviour of `BehaviourSubject` is weird when a `scan()`'s involved (`scan` only accumulates after you've subscribed):

```js
const subject = new rxjs.BehaviorSubject(0).scan((acc, val) => acc + val, 0)
subject.subscribe(console.log) // prints 0
subject.next(1) // prints 1
subject.next(1) // prints 2
subject.subscribe(console.log) // prints 2, 1
subject.next(1) // prints 3, 2
```

Using `publishReplay()` seems to solve this:

```js
const subject = new rxjs.BehaviorSubject(0).scan((acc, val) => acc + val, 0).publishReplay(1)
subject.connect()
subject.subscribe(console.log) // prints 0
subject.next(1) // prints 1
subject.next(1) // prints 2
subject.subscribe(console.log) // prints 2, 2
subject.next(1) // prints 3, 3
```

Reply:
Weird! Nice catch :)